programs/lighthouse/src/types/assert/evaluate.rs

```diff
@@ -119,32 +36,14 @@ macro_rules! impl_integer_evaluate {
                         IntegerOperator::DoesNotContain => actual_value & assertion_value == 0,
                     };
 
-                    match log_level {
-                        LogLevel::PlaintextMessage => {
-                            msg!(
-                                "Result: {} {} {}",
-                                actual_value,
-                                operator.format(),
-                                assertion_value
-                            );
-                        }
-                        LogLevel::Silent => {}
-                        LogLevel::EncodedMessage => {
-                            AssertionResult::$payload_variant(
-                                Some(*actual_value),
-                                Some(*assertion_value),
-                                *operator as u8,
-                                passed,
-                            ).log_data()?;
-                        }
-                        LogLevel::EncodedNoop => {
-                            AssertionResult::$payload_variant(
-                                Some(*actual_value),
-                                Some(*assertion_value),
-                                *operator as u8,
-                                passed,
-                            ).log_noop()?;
-                        }
+                    if log_level != LogLevel::Silent {
```

Comment:
You don't need the guard right? If you mark `AssertionResult::log` as inline it should be the same cost anyways

Reply:
The cost is all the values need to be copied into AssertionResult and if you don't want anything logged than thats wasted compute.